### PR TITLE
testlapack: improve test for Dlahqr

### PR DESCRIPTION
**Pull request by vladimir-ch**

### CODE REVIEW

**Comment by vladimir-ch** [comment]
_ping_ (I use the helper functions for testing Dlaqr2)

File: testlapack/general.go [hunk 1]
```diff
@@ -950,3 +951,47 @@ func schurBlockSize(t blas64.General, i int) (size int, first bool) {
...
+func isAllNaN(x []float64) bool {
+	for _, v := range x {
+		if !math.IsNaN(v) {
+			return false
+		}
+	}
+	return true
+}
+
+func isAnyNaN(x []float64) bool {
```
Comment (kortschak) [inline testlapack/general.go]: Use floats.HasNaN.

**Comment by kortschak** [comment]
LGTM but wait for @btracey

File: testlapack/general.go [hunk 2]
```diff
@@ -950,3 +951,47 @@ func schurBlockSize(t blas64.General, i int) (size int, first bool) {
...
+	for _, v := range x {
+		if math.IsNaN(v) {
+			return true
+		}
+	}
+	return false
+}
+
+func isHessenberg(h blas64.General) bool {
```
Comment (btracey) [inline testlapack/general.go]: isUpperHessenberg?

**Pull request merged**